src/connect4.ts

```diff
@@ -120,9 +126,15 @@ export function checkBoardStateConsistency(boardState: BoardState): void {
 /**
  * @throws SyntaxError if the board is invalid (checkBoardStateConsistency)
  */
-export function initGameState(stateConfigFile: BoardState): GameState {
+export function initGameState(stateConfigFile: BoardState | null): GameState {
+  const boardState: BoardState = stateConfigFile
+    ? stateConfigFile
+    : Array.from({ length: Number(boardLayout.NB_ROWS) }, () =>
+        Array(boardLayout.NB_COLUMN).fill(0),
+      );
```

Comment:
```suggestion
    : Array.from({ length: Number(boardLayout.NB_ROWS) }, () =>
        Array(boardLayout.NB_COLUMN).fill(0),
      );
```

You can extract it to a new (and tested) function